### PR TITLE
po: document process to generate launchpad translations

### PR DESCRIPTION
**Pull request by ZeyadYasser**
This README is extracted from an older attempt to sync snapd launchpad translations which are now outdated but the guide is still relevant.

Reference: https://github.com/snapcore/snapd/pull/12912

### CODE REVIEW

File: po/README.md [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+- Export latest translations [here](https://translations.launchpad.net/snapd/trunk/+export) (download link will be sent to your email)
+- Extract archive: `tar -xvf launchpad-export.tar.gz`
+- Create new directory to store processed translations `mkdir processed`
+- `for file in snappy-*; do msgattrib --no-obsolete --translated --output-file="processed/${file#snappy-}" "$file"; done;`
+    - Removes the `snappy-` prefix
+    - `--no-obsolete` removes obsolete #~ messages
+    - `--translated` excludes untranslated messages
+- Replace old translations under `/po` with new translations
+- Commit author and message should match Launchpad automated commits
+    - git commit --author="Launchpad Translations on behalf of snappy-dev[bot]<noreply@canonical.com>" -m "po: launchpad automatic translations update"
```
Comment (MiguelPires) [inline po/README.md:16]: Doesn't this clash against the cla check?

Comment (ZeyadYasser) [reply po/README.md:16]: Good point, I will remove this line for now. The future plan is to actually create a bot to do this periodically.

File: po/README.md [hunk 2]
```diff
@@ -0,0 +1,19 @@
+# Translations generated by Launchpad
+
+## How to sync translations from Launchpad?
+- Run `update-pot` to generate latest translation template (`po/snappy.pot`)
+- Import generated translation template to snappy entry [here](https://translations.launchpad.net/snapd/trunk/+templates)
+- Wait until import is approved. You can monitor status in the import queue [here](https://translations.launchpad.net/snapd/trunk/+imports)
+- Export latest translations [here](https://translations.launchpad.net/snapd/trunk/+export) (download link will be sent to your email)
```
Comment (ernestl) [inline po/README.md:7]: Consider using this cleaner way of referencing urls: https://github.com/snapcore/snapd/blob/master/CONTRIBUTING.md?plain=1#L161-L162

**Review by ernestl — COMMENTED** [review]
Thanks for this, some nitpicks for consideration

File: po/README.md [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+- Extract archive: `tar -xvf launchpad-export.tar.gz`
+- Create new directory to store processed translations `mkdir processed`
+- `for file in snappy-*; do msgattrib --no-obsolete --translated --output-file="processed/${file#snappy-}" "$file"; done;`
+    - Removes the `snappy-` prefix
+    - `--no-obsolete` removes obsolete #~ messages
+    - `--translated` excludes untranslated messages
+- Replace old translations under `/po` with new translations
+
+References:
```
[the remark this answers is not in the document]
Comment (ZeyadYasser) [reply po/README.md]: Good point, thanks! updated

File: po/README.md [hunk 4]
```diff
@@ -0,0 +1,19 @@
+# Translations generated by Launchpad
+
+## How to sync translations from Launchpad?
+- Run `update-pot` to generate latest translation template (`po/snappy.pot`)
```
[the remark this answers is not in the document]
Comment (ZeyadYasser) [reply po/README.md]: Nice catch, I already had those set, updated the document to explicitly mention it. Thanks!

File: po/README.md (hunk 2 above)
Comment (ZeyadYasser) [reply po/README.md]: updated

**Review by ernestl — APPROVED** [review]
Thanks

**Review by MiguelPires — APPROVED** [review]

File: po/README.md [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+
+## How to sync translations from Launchpad?
+- Make sure `GOPATH` is set and `GOINVOKEFLAGS="-mod=readonly"`.
+- Run `update-pot` to generate latest translation template (`po/snappy.pot`).
+- Import generated translation template to [the snappy entry][templates].
+- Wait until import is approved. You can monitor status in the [import queue][import-queue].
+- Export [latest translations][export] (download link will be sent to your email).
+- Extract archive: `tar -xvf launchpad-export.tar.gz`.
+- Create new directory to store processed translations `mkdir processed`.
+- `for file in snappy-*; do msgattrib --no-obsolete --translated --output-file="processed/${file#snappy-}" "$file"; done;`
```
Comment (degville) [inline po/README.md:11]: Dropping straight to a script like this seems a little jarring. Perhaps prefix this line with something like:
- Run the following to remove the `snappy-` prefix, remove obsolete `#-` messages (`--no-obsolete`) and exclude untranslated messages (`--translated`):
 `for file in snappy-*; do msgattrib --no-obsolete --translated --output-file="processed/${file#snappy-}" "$file"; done;`

**Review by degville — APPROVED** [review]
Looks good to me - thank you! One small comment which isn't important.

File: po/README.md (hunk 5 above)
Comment (ZeyadYasser) [reply po/README.md:11]: Great point, Thank you!

**Pull request merged**